README.md

```diff
@@ -1404,6 +1404,60 @@ BlockPayload.batch_with_outputs(payload_hashes)
    ]
  }}
 ```
+
+### Mempool P2P Endpoints
+Mempool P2P endpoints for communication between mempools. Endusers are not supposed to use these endpoints directly. Instead, the respective Pact endpoints should be used for submitting transactions into the network.
+
+#### Get Pending Transactions from the Mempool
+
+```elixir
+Kadena.Chainweb.P2P.Mempool.retrieve_pending_txs(network_opts \\ [])
+```
+
+**Parameters**
+
+- `network_opts`: Network options. Keyword list with:
+  - `network_id` (required): Allowed values: `:testnet04` `:mainnet01`.
+  - `location` (optional): Location to access a Chainweb P2P bootstrap node. Allowed values:
+    - testnet: `"us1"`, `"us2"`, `"eu1"`, `"eu2"`, `"ap1"`, `"ap2"`
+    - mainnet: `"us-e1"`, `"us-e2"`, `"us-e3"`, `"us-w1"`, `"us-w2"`, `"us-w3"`, `"fr1"`, `"fr2"`, `"fr3"`, `"jp1"`, `"jp2"`, `"jp3"`
+  - `chain_id` (required): Id of the chain to which the request is sent. Allowed values: integer or string-encoded integer from 0 to 19.
+  - `query_params` (optional): Query parameters. Keyword list with:
+    - `nonce`: Integer value. Server nonce value.
+    - `since`: Integer value. Mempool tx id value.
+
+  Defaults to `[network_id: :testnet04, location: "us1", chaind_id: 0]` if not specified. If `network_id` is set as `:mainnet01` the default `location` is `"us-e1"`
+  
+```elixir
+alias Kadena.Chainweb.P2P.Mempool
+
+Mempool.retrieve_pending_txs(network_id: :mainnet01)
+
+{:ok,
+ %Kadena.Chainweb.P2P.MempoolRetrieveResponse{
+   hashes: [
+     "jCNv2dvA0tLUmxJBmIkcgclln1-slQwd_T2rlHmKtWM",
+     "zStNSKpuc1vvzRHKlYh7H9Az2CqFN5He4zqhf_t7AxQ",
+     "bsslUCR93WCwQ7VT4YLkTCjdah7pdFkOkl2zShZdtb8",
+     "SIHfvd0R5Dm4ZjIRDu4mrQJwByCwi2H23PlyJpBoF2o",
+     "6E2Ak5BfG7-i-5_pedS2Rs0GH4vX8dXdB6b1knCFSVE",
+     "6RalAOOgJwVtJtYVmoT2t8V1VbuEawp0lrXNvRubqOo",
+     "6ZqWX6l16nmgK55v6EdMDLEbnR9Ie2ihealJM4QVvI8",
+     "avjoE1NxEAYEIOLPucS-sjmw-JqVfjUOx6Cxw0cjNwk",
+     "C385m6e9S7WzelUFCyW-JoZFJGQNlcI0jqCO8YrPMVo",
+     "hK1dutkawvL5Pt79rMzA8JnQZyUesAY0ce8XL0sHIqc",
+     "9HgO9Q47EcQFfNggaeio43KXqR1acezyxzMFZxavnjc",
+     "BSyDvmN4varpyVrxGUiy6rBKhZe4QkNGD8MLWQvA-MI",
+     "ldyt5_FB6-CwM3G2ccGyU4eeWsMVpg0yDDd3jtY_tjs",
+     "JrK3340Q513CHrdZ-EIdUa7nHu4WJSc1GtHn-xkBUSs",
+     "QHUIyBa0kUwF8WwOO_HSxdhVsh6zYu3LllFJeNEndGI",
+     "wo3iU8tm4Q3vIcK3-4UB19TV0LXideuJ7OQK09axbTs",
+     "0lzHjlVHMmqFyLAFjW8R4osVS2xE4_SNu5KwtYwLSps"
+   ],
+   highwater_mark: [-2_247_324_167_920_489_014, 71414]
+ }}
```

Comment:
I think it would be valuable for the example to use the query parameters (nonce, since)